From 06f80305f342351b36f3bd8eb38ea0c634781a60 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Tue, 7 Oct 2025 20:55:24 +0300
Subject: [PATCH 349/553] tst_QJson: port away from QPair

Pick-to: 6.8 6.5
Task-number: QTBUG-115841
Change-Id: If392bd54f8e4eff139fa02f88620a1170f8fdb59
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 9b1a368622c411ffa574de2373c64782dfa21778)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/corelib/serialization/json/tst_qtjson.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/tests/auto/corelib/serialization/json/tst_qtjson.cpp b/tests/auto/corelib/serialization/json/tst_qtjson.cpp
index 127c8228616..2e0d8dd7be2 100644
--- a/tests/auto/corelib/serialization/json/tst_qtjson.cpp
+++ b/tests/auto/corelib/serialization/json/tst_qtjson.cpp
@@ -18,6 +18,8 @@ QT_WARNING_DISABLE_DEPRECATED
 #include "private/qjson_p.h"
 #include <limits>
 
+using namespace Qt::StringLiterals;
+
 #define INVALID_UNICODE "\xCE\xBA\xE1"
 #define UNICODE_NON_CHARACTER "\xEF\xBF\xBF"
 #define UNICODE_DJE "\320\202" // Character from the Serbian Cyrillic alphabet
@@ -4282,7 +4284,7 @@ void tst_QtJson::fromToVariantConversions()
     QVariant variantFromJsonArray(QJsonArray { json });
     QVariant variantFromJsonObject(QVariantMap { { "foo", variant } });
 
-    QJsonObject object { QPair<QString, QJsonValue>("foo", json) };
+    QJsonObject object = { {"foo", json} };
 
     // QJsonValue <> QVariant
     {
@@ -4326,8 +4328,8 @@ void tst_QtJson::fromToVariantConversions()
         // test the same for QVariantMap from QJsonValue/QJsonArray/QJsonObject
         QCOMPARE(QJsonObject::fromVariantMap(QVariantMap { { "foo", variantFromJson } }), object);
 
-        QJsonObject nestedArray { QPair<QString, QJsonArray>("bar", QJsonArray { json }) };
-        QJsonObject nestedObject { QPair<QString, QJsonObject>("bar", object) };
+        QJsonObject nestedArray = { {u"bar"_s, QJsonArray{json}} };
+        QJsonObject nestedObject = { {u"bar"_s, object } };
         QCOMPARE(QJsonObject::fromVariantMap(QVariantMap { { "bar", variantFromJsonArray } }),
                  nestedArray);
         QCOMPARE(QJsonObject::fromVariantMap(QVariantMap { { "bar", variantFromJsonObject } }),
-- 
2.51.2

